pkg/backup/item_collector.go

```diff
@@ -287,18 +292,50 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
 		if selector := r.backupRequest.Spec.LabelSelector; selector != nil {
 			labelSelector = metav1.FormatLabelSelector(selector)
 		}
+		listOptions := metav1.ListOptions{LabelSelector: labelSelector}
 
 		log.Info("Listing items")
-		unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
-		if err != nil {
-			log.WithError(errors.WithStack(err)).Error("Error listing items")
-			continue
+		unstructuredItems := make([]unstructured.Unstructured, 0)
+
+		if r.pageSize == 0 {
+			// If limit is the special value 0, do not use paging. Instead, request all items at once
+			unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
+			unstructuredItems = append(unstructuredItems, unstructuredList.Items...)
+			if err != nil {
+				log.WithError(errors.WithStack(err)).Error("Error listing items")
+				continue
+			}
+		} else {
+			// If limit is positive, use a pager to split list over multiple requests
+			// Use Velero's dynamic list function instead of the default
+			listFunc := pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
+				list, err := resourceClient.List(listOptions)
```

Comment:
if you are not planning on checking the type of error, simply `return resourceClient.List(listOptions)` might work as well.

Also, can you move the TODO on line 320 here? I think the `ResourceExperied` errors will be handled here in the future.

Reply:
Page buffer size would be handled by setting `listPager.PageBufferSize`.

I double checked how `ResourceExpired` errors need to be handled and it looks like `ListPager` only handles them for us if we use `List()` instead of `EachListItem()` :( So this needs to be changed to handle the error after all.